tuna/worker_interface.py

```diff
@@ -270,43 +298,30 @@ def get_job(self, find_state, set_state, imply_end):
     return False
 
   # JD: This should take a session obj as an input to remove the creation of an extraneous session
-  def set_job_state(self, state, increment_retries=False, result=''):
+  def set_job_state(self, state, increment_retries=False, result=None):
     """Interface function to update job state for builder/evaluator"""
     self.logger.info('Setting job id %s state to %s', self.job.id, state)
     for idx in range(NUM_SQL_RETRIES):
       with DbSession() as session:
         try:
-          if state in ["running", "compiling", "evaluating"]:
-            session.query(self.dbt.job_table).filter(
-                self.dbt.job_table.id == self.job.id).update({
-                    self.dbt.job_table.state: state,
-                    self.dbt.job_table.result: result
-                })
-          else:
-            if increment_retries:
-              session.query(self.dbt.job_table).filter(
-                  self.dbt.job_table.id == self.job.id).update({
-                      self.dbt.job_table.state:
-                          state,
-                      self.dbt.job_table.retries:
-                          self.dbt.job_table.retries + 1,
-                      self.dbt.job_table.result:
-                          result
-                  })
-            else:
-              # JD: When would this happen ?
-              # also this is a side-effect, not cool
-              cache = '~/.cache/miopen_'
-              blurr = ''.join(
-                  random.choice(string.ascii_lowercase) for i in range(10))
-              cache_loc = cache + blurr
-              session.query(self.dbt.job_table).filter(
-                  self.dbt.job_table.id == self.job.id).update({
-                      self.dbt.job_table.state: state,
-                      self.dbt.job_table.cache_loc: cache_loc,
-                      self.dbt.job_table.result: result
-                  })
+          #refresh job data
+          self.job = session.query(self.dbt.job_table)\
+              .filter(self.dbt.job_table.id == self.job.id).one()
+          self.job.state = state
+          if result:
+            self.job.result = result
+          if increment_retries:
+            self.job.retries += 1
+
+          if '_start' in state:
+            cache = '~/.cache/miopen_'
```

Comment:
hmm maybe we can set this location based on the library enum not hardcode the miopen?

Reply:
A fine idea, but not in scope for this pr.